### ISSUE TITLE
Make it easier to determine vendor dir location

### ISSUE THREAD

**Issue by hypeJunction** [Elgg/Elgg#9006]
This is currently what I have to do to determine vendor dir path. I think we need simplify this. Can't the views registration service just iterate through all locations and see if the file we are registering is there?

https://github.com/hypeJunction/elgg_tooltip/blob/master/views.php

``` php
$plugin_root = __DIR__;
$root = dirname(dirname($plugin_root));
$alt_root = dirname(dirname(dirname($initial_root)));
if (file_exists("$plugin_root/vendor/autoload.php")) {
    $path = $plugin_root;
} else if (file_exists("$root/vendor/autoload.php")) {
    $path = $root;
} else {
    $path = $alt_root;
}
return array(
    'default' => array(
        'tooltipster.js' => $path . '/vendor/bower-asset/tooltipster/js/jquery.tooltipster.min.js',
        'tooltipster.css' => $path . '/vendor/bower-asset/tooltipster/css/tooltipster.css',
    )
);
```

**Comment by mrclay** [comment]
Have you tried:

```php
return array(
    'default' => array(
        'tooltipster.js' => 'vendor/bower-asset/tooltipster/js/jquery.tooltipster.min.js',
        'tooltipster.css' => 'vendor/bower-asset/tooltipster/css/tooltipster.css',
    )
);
```

**Comment by mrclay** [comment]
Since these are in a plugin, why can't you use `__DIR__ . '/vendor/...`?

**Comment by hypeJunction** [comment]
In my projects I just require my plugins from root. I suppose that's what you would do with a starter project as well.

**Comment by mrclay** [comment]
Are there other cases besides these?
- the plugin is a composer dependency of the root project
- a plugin uses composer locally

**Comment by hypeJunction** [comment]
the plugins is a composer dependency of a the starter project

**Comment by mrclay** [comment]
> Can't the views registration service just iterate through all locations and see if the file we are registering is there?

I guess if a path starts with "vendor/" we could do this, though the worry is checking plugin > Elgg > root may not always be the desired strategy. I'd imagine you'd instead want root > plugin > Elgg.

**Comment by jeabakker** [comment]
I believe this was fixed in #11106

The usage pattern now can be
```php
$path = '';
if (is_dir(__DIR__ . '/vendor/') {
     $path = __DIR__ . '/';
}

return array(
    'default' => array(
        'tooltipster.js' => $path . 'vendor/bower-asset/tooltipster/js/jquery.tooltipster.min.js',
        'tooltipster.css' => $path . 'vendor/bower-asset/tooltipster/css/tooltipster.css',
    )
);
```
the relative path (first case) handles your plugin being installed through the `composer.json` of the starter project or the `composer.json` of Elgg (if that is the root of your project).
the absolute path (second case) handles plugin installed on it's own (probably a zip from the Elgg community)
